tests/components/vizio/test_media_player.py

```diff
@@ -94,6 +132,72 @@
             )
 
 
+async def _test_setup_with_apps(
+    hass: HomeAssistantType, device_config: Dict[str, Any], app: str
+) -> None:
+    """Test Vizio Device with apps entity setup."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN, data=vol.Schema(VIZIO_SCHEMA)(device_config), unique_id=UNIQUE_ID
+    )
+
+    with patch(
+        "homeassistant.components.vizio.media_player.VizioAsync.get_all_audio_settings",
+        return_value={
+            "volume": int(MAX_VOLUME[VIZIO_DEVICE_CLASS_TV] / 2),
+            "mute": "Off",
+        },
+    ), patch(
+        "homeassistant.components.vizio.media_player.VizioAsync.get_power_state",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.vizio.media_player.VizioAsync.get_current_app",
+        return_value=app,
+    ), patch(
+        "homeassistant.components.vizio.media_player.VizioAsync.get_current_app_config",
+        return_value=AppConfig(**ADDITIONAL_APP_CONFIG["config"]),
+    ):
+        config_entry.add_to_hass(hass)
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+        attr = hass.states.get(ENTITY_ID).attributes
+        assert attr["friendly_name"] == NAME
+        assert attr["device_class"] == DEVICE_CLASS_TV
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5141152)